source/particle/world.cc

```diff
@@ -158,19 +158,36 @@ namespace aspect
     {
       // If we do not write output
       // return early with the number of particles that were advected
-      if (!output)
+      if (output.empty())
         return "";
 
       TimerOutput::Scope timer_section(this->get_computing_timer(), "Particles: Output");
       const double output_time = (this->convert_output_to_years() ?
                                   this->get_time() / year_in_seconds :
                                   this->get_time());
 
-      const std::string filename = output->output_particle_data(particles,
-                                                                property_manager->get_data_info(),
-                                                                output_time);
+      std::vector<std::string> output_suffixes;
 
-      return filename;
+      const std::string output_directory = output[0].get()->get_particle_output_location();
+      const std::string file_index = output[0].get()->get_file_index();
+
+      for ( typename std::vector<std_cxx11::shared_ptr<Output::Interface<dim> > >::const_iterator itr= output.begin(); itr != output.end(); itr++)
```

Comment:
use `++itr` unless there is need for `itr++` (almost never, unless you modify the container within the loop). If you are interested in the topic here is a somewhat lengthy discussion: https://stackoverflow.com/questions/484462/difference-between-i-and-i-in-a-loop?noredirect=1&lq=1).